app/forms/NiphlsNumberFormProvider.scala

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms
+
+import forms.mappings.Mappings
+import models.NiphlsNumber
+import play.api.data.Form
+
+class NiphlsNumberFormProvider extends Mappings {
+
+  def apply(): Form[String] = Form(
+    "value" -> text("niphlsNumber.error.notSupplied")
+      .verifying(
+        regexp("^([0-9]{4,6}|[a-zA-Z]{1,2}[0-9]{5})$", "niphlsNumber.error.wrongFormat")
```

Comment:
i feel like this should be a model like : https://github.com/hmrc/trader-goods-profiles-frontend/pull/13/files

Reply:
It's a model now 🙂